avocado/plugins/run.py

```diff
@@ -77,7 +77,10 @@ def configure(self, parser):
                                   'You can also use suffixes, like: '
                                   ' s (seconds), m (minutes), h (hours). '))
 
-        parser.add_argument('--failfast', action='store_true',
+        parser.add_argument('--failfast',
+                            nargs='?',
+                            const=True,
+                            type=self._failfast,
                             help='Interrupt job on first failed test.')
```

Comment:
Btw I'm not much in favor of optional arguments for this, did you consider using `--without-failfast` and implement it as `store_false` with `target=failfast`? In my eyes that would look better. What do you think? (I mean both versions are acceptable by me, I'd just rather see `--without-failfast` than this optional arguments...)